packages/gitbook/src/app/middleware/(site)/(content)/layout.tsx

```diff
@@ -22,14 +23,14 @@ export default async function ContentLayout(props: { children: React.ReactNode }
 
     const context = await fetchLayoutData();
     const queryStringTheme = await getThemeFromMiddleware();
-    const visitorAuthToken = await getVisitorAuthTokenFromMiddleware();
+    const siteData = await getSiteURLDataFromMiddleware();
 
     return (
         <SiteLayout
             context={context}
             forcedTheme={queryStringTheme}
             withTracking={await shouldTrackEvents()}
```

Comment:
Sure we don't want `Promise.all` here?

Reply:
yes, it has no impact on perfs, it's just a `await headers()`, Next uses `await as a way to detect the calls and trigger the route to be dynamic